TFT/src/User/API/Temperature.c

```diff
@@ -15,17 +15,30 @@ static bool    heat_send_waiting[MAX_HEATER_COUNT];
 uint32_t nextHeatCheckTime = 0;
 #define AUTOREPORT_TIMEOUT (nextHeatCheckTime + 3000)  // update interval + 3 second grace period
 
-static uint8_t fixHeaterIndex(uint8_t index)
+// Verify that the heater index is valid, and fix the index of multiple in and 1 out tool nozzles
+static uint8_t heaterIndexFix(uint8_t index)
 {
-  if (infoSettings.hotend_count == 1)
-    index = (index < MAX_HOTEND_COUNT) ? NOZZLE0 : index;
-  return index;
+  if (index == BED && infoSettings.bed_en)  // Bed
+    return index;
+  if (index == CHAMBER && infoSettings.chamber_en)  // Chamber
+    return index;
+  if (index < infoSettings.hotend_count)  // Vaild tool nozzle
+    return index;
+  if (index < infoSettings.ext_count && infoSettings.hotend_count == 1)  // "multi-extruder" that shares a single nozzle.
+  {
+    return NOZZLE0;
+  }
```

Comment:
The logic will never reach to line number 27 (`if (index < infoSettings.ext_count && infoSettings.hotend_count == 1)`) because the index passed will always be less than infoSettings.hotend_count.